### PR TITLE
Add user data to wlr_* structs

### LINKED ISSUE #68: Add user data storage mechanism to wlr_* types
We should let the user attach something to `wlr_*` types, especially after moving wl_shell/xdg_shell/etc into wlr.

### PR DESCRIPTION
**Pull request by 4e554c4c**
Resolves #68

### CODE REVIEW

File: types/wlr_keyboard.c [hunk 1]
```diff
@@ -9,6 +9,7 @@ struct wlr_keyboard *wlr_keyboard_create(struct wlr_keyboard_impl *impl,
 	struct wlr_keyboard *kb = calloc(1, sizeof(struct wlr_keyboard));
 	kb->impl = impl;
 	kb->state = state;
+	kb->data = NULL;
```
Comment (SirCmpwn) [inline types/wlr_keyboard.c]: This sort of thing is not necessary since we use calloc

File: examples/compositor/wl_compositor.c [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 static void destroy_surface_listener(struct wl_listener *listener, void *data) {
 	struct wlr_surface *surface = wl_resource_get_user_data(data);
-	struct wl_compositor_state *state = surface->compositor_data;
```
Comment (SirCmpwn) [inline examples/compositor/wl_compositor.c]: Hold off on this change

Comment (4e554c4c) [inline examples/compositor/wl_compositor.c]: Should I create both a `compositor_data` and `data` field in wlr_surface then? I think it's harmless until wl_compositor gets merged into wlroots. Also I created #71 for this

Comment (SirCmpwn) [inline examples/compositor/wl_compositor.c]: Yes, leave it how it was before, with special fields in wlr_surface for the compositor.

File: types/wlr_keyboard.c (hunk 1 above)
Comment (4e554c4c) [inline types/wlr_keyboard.c]: Thanks, I didn't notice this

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**